crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -88,6 +91,25 @@ fn fragment(in: FragmentInput) -> @location(0) vec4<f32> {
         pbr_input.material.metallic = metallic;
         pbr_input.material.perceptual_roughness = perceptual_roughness;
 
+        var transmission: f32 = material.transmission;
+        if ((material.flags & STANDARD_MATERIAL_FLAGS_TRANSMISSION_TEXTURE_BIT) != 0u) {
+            transmission *= textureSample(transmission_texture, transmission_sampler, uv).r;
+        }
+        pbr_input.material.transmission = transmission;
+
+        var thickness: f32 = material.thickness;
+        if ((material.flags & STANDARD_MATERIAL_FLAGS_THICKNESS_TEXTURE_BIT) != 0u) {
+            thickness *= textureSample(thickness_texture, thickness_sampler, uv).g;
+        }
+        thickness *= (length(mesh.model[0].xyz) + length(mesh.model[1].xyz) + length(mesh.model[2].xyz)) / 3.0;
```

Comment:
It's calculating the average of the basis vectors of the transform matrix, so that we can figure out how much to "scale" the thickness.

Ideally we would get the local normal at that point, multiply by the transform matrix and measure the length of that to figure out how much to scale the thickness at that specific point/direction, however AFAIK the local normal is not passed to the fragment shader.

Non-uniform scaling like stretching the model in one axis only will produce somewhat wrong results, but I feel like this is a rare enough case that it's not worth it to add the plumbing needed to do the right thing.